### PR TITLE
Add test for `athena-framework`

### PR DESCRIPTION
**Pull request by straight-shoota**

### CODE REVIEW

**Comment by straight-shoota** [comment]
The point of this repository is to gauge the effect of changes on important code bases in the ecosystem. It allows us to run tests with specific development builds of the compiler and stdlib.
This goes in parallel to test setups of the individual projects which cannot easily be triggered like this.

**Review by beta-ziliani — APPROVED** [review]

File: test/athena.bats [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  load helper/common.bash
+}
+
+function test_component() {
+  ./scripts/test.sh "$1"
+}
+
+@test "config" {
+  test_component "config"
+}
```
Comment (Blacksmoke16) [inline test/athena.bats:21]:
```suggestion
@test "clock" {
  test_component "clock"
}
```

The config component is now defunct, but `clock` was recently added.

**Review by Blacksmoke16 — APPROVED** [review]

File: test/athena.bats [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+@test "event_dispatcher" {
+  test_component "event_dispatcher"
+}
+@test "framework" {
+  test_component "framework"
+}
+@test "image_size" {
+  test_component "image_size"
+}
```
Comment (Blacksmoke16) [inline test/athena.bats:39]:
```suggestion
@test "image_size" {
  test_component "image_size"
}
@test "mercure" {
  test_component "mercure"
}
```
This is a fairly new component.

**Comment by straight-shoota** [comment]
The shards install failure is weird: https://github.com/crystal-lang/test-ecosystem/actions/runs/11276440947/job/31360217345?pr=56

```
    I: Resolving dependencies
   I: Fetching https://github.com/athena-framework/framework.git
   I: Fetching https://github.com/athena-framework/clock.git
   I: Fetching https://github.com/athena-framework/console.git
   I: Fetching https://github.com/athena-framework/dependency-injection.git
   I: Fetching https://github.com/athena-framework/dotenv.git
   I: Fetching https://github.com/athena-framework/event-dispatcher.git
   I: Fetching https://github.com/athena-framework/image-size.git
   I: Fetching https://github.com/athena-framework/mercure.git
   I: Fetching https://github.com/athena-framework/negotiation.git
   I: Fetching https://github.com/athena-framework/routing.git
   I: Fetching https://github.com/athena-framework/serializer.git
   I: Fetching https://github.com/athena-framework/spec.git
   I: Fetching https://github.com/athena-framework/validator.git
   I: Fetching https://github.com/crystal-ameba/ameba.git
   I: Fetching https://github.com/athena-framework/spec.git
   E: Failed to clone https://github.com/athena-framework/spec.git: Failed git clone -c core.askPass=true -c init.templateDir= --mirror --quiet -- https://github.com/athena-framework/spec.git /home/runner/.cache/shards/github.com/athena-framework/spec.git.
   fatal: destination path '/home/runner/.cache/shards/github.com/athena-framework/spec.git' already exists and is not an empty directory.
```

Reproduces on Linux and macOS, but not on Windows. I also cannot reproduce on Linux locally.
Is this a bug in shards? I notice there are two lines `Fetching https://github.com/athena-framework/spec.git`, maybe shards got into a confused state?

**Comment by Blacksmoke16** [comment]
Is this tool intended to run against the lastest released version, or the latest code in master?

**Comment by straight-shoota** [comment]
> Is this tool intended to run against the lastest released version, or the latest code in master?

It runs against master. That's usually best because in case there are some incompatibilities with the current Crystal master, they might be fixed but not released yet.

**Comment by Blacksmoke16** [comment]
Alright, in that case I'd prob just do like `SHARDS_OVERRIDE=shard.dev.yml shards install`. That'll just symlink everything from `src/` into `lib/`. Which is what I do for Athena CI as well.

**Comment by straight-shoota** [comment]
That fixed it. Still something is odd in the `shard.yml`.

And there's a spec failure in `spec` on Windows.

**Comment by Blacksmoke16** [comment]
> Still something is odd in the `shard.yml`.

Is prob because it's both a dev and non-dev dependency? 

> And there's a spec failure in `spec` on Windows

I think what's going on here is Athena specs have 2 diff "types" `unit` and `compiled`. The latter being responsible for testing things that run via `Process.run`. In normal Athena CI, the compiled specs only run on ubuntu, because there's not really any platform specific logic that really needs tested.

So I guess to fix this, I could either:

1. Mark this spec as pending if not on a unix like OS
2. Have this repo only run compiled specs on linux
3. Make this spec less specific by just looking at the exception type vs the exact message

3 sounds pretty reasonable, I'll put together a PR and get merged then can re-run that spec.

**Comment by straight-shoota** [comment]
> > Still something is odd in the `shard.yml`.
>
> Is prob because it's both a dev and non-dev dependency? 

That shouldn't be an issue.
Also it works locally. So there's definitely something quirky going on.

**Comment by Blacksmoke16** [comment]
@straight-shoota This looks good now.

**Pull request merged**